FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureBlobJobStore.cs

```diff
@@ -199,6 +229,11 @@ public async Task CompleteJobAsync(Job job, CancellationToken cancellationToken
                 // Remove job from active job list.
                 await _blobContainerClient.DeleteBlobAsync(GetJobBlobName(job, AzureBlobJobConstants.ActiveJobFolder), cancellationToken);
 
+                if (releaseLock)
+                {
+                    await TryReleaseJobLockAsync(cancellationToken);
```

Comment:
If encounter error/exception before release job lock, do we need to manually release the blob lock before retry or restart the pipeline agent?

Reply:
Yes. Otherwise the next trigger may encounter conflicts. That's a good point. I have added the release logic for exceptions in job execution